cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/TypeManager.kt

```diff
@@ -187,13 +190,35 @@ class TypeManager {
         return parameterizedType
     }
 
-    fun <T : Type> registerType(t: T): T {
+    inline fun <reified T : Type> registerType(t: T): T {
+        // Skip as they should be unique to each class and not globally unique
+        if (t is ParameterizedType) {
+            return t
+        }
+
         if (t.isFirstOrderType) {
-            firstOrderTypes.add(t)
-        } else {
-            secondOrderTypes.add(t)
-            registerType((t as SecondOrderType).elementType)
+            // Make sure we only ever return one unique object per type
+            if (!firstOrderTypes.add(t)) {
```

Comment:
Could probably be optimized by using a Map and `computeIfAbsent()`

Reply:
I tried that and its not possible because it only returns the value of the map which is then `True`, but I optimised it a little bit by only using `first` and not `filterIsInstance` and `first`.